src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,114 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { client } from '../utils/fetchClient';
+
+type Props = {
+  selectedPost: Post | null,
+  setPostsComments: React.Dispatch<React.SetStateAction<Comment[]>>,
+  setCommentErrorMessage: (value: string) => void,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  setPostsComments,
+  setCommentErrorMessage,
+}) => {
+  const [isFormLoading, setIsFormLoading] = useState(false);
+  const [commentName, setCommentName] = useState('');
+  const [commentEmail, setCommentEmail] = useState('');
+  const [commentText, setCommentText] = useState('');
+  const [formError, setFormError] = useState({
+    name: '',
+    email: '',
+    text: '',
+  });
+
+  const isEmptyField = !!commentName.length
+    || !commentEmail.length
+    || !commentText.length;
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCommentName(event.target.value);
+    setFormError({ ...formError, name: '' });
+  };
```

Comment:
you may save all values in state as object (the same as errors), and create 1 handler. Use event.target.name as key for object:

```
setFormValue(prevValues => ({ ...prevValues, [name]: value }))
```